addons/binding/org.openhab.binding.windcentrale/src/main/java/org/openhab/binding/windcentrale/handler/WindCentraleHandler.java

```diff
@@ -0,0 +1,156 @@
+            @Override
+            public void run() {
+                updateData();
+            }
+        }, 0, pollingPeriod, TimeUnit.SECONDS);
+        logger.debug("Polling job scheduled to run every {} sec. for '{}'", pollingPeriod, getThing().getUID());
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing WindCentrale handler '{}'", getThing().getUID());
+        if (pollingJob != null) {
+            pollingJob.cancel(true);
+            pollingJob = null;
+        }
+    }
+
+    private synchronized void updateData() {
+        logger.debug("Update WindMolen data '{}'", getThing().getUID());
+
+        try {
+            String getMillData = getMillData();
+            JsonParser parser = new JsonParser();
+            JsonObject millData = (JsonObject) parser.parse(getMillData);
+            logger.debug(millData.toString());
+            updateState(CHANNEL_WIND_SPEED, new DecimalType(millData.get(CHANNEL_WIND_SPEED).getAsString()));
+            updateState(CHANNEL_WIND_DIRECTION, new StringType(millData.get(CHANNEL_WIND_DIRECTION).getAsString()));
+            updateState(CHANNEL_POWER_TOTAL, new DecimalType(millData.get(CHANNEL_POWER_TOTAL).getAsBigDecimal()));
+            updateState(CHANNEL_POWER_PER_WD,
+                    new DecimalType(millData.get(CHANNEL_POWER_PER_WD).getAsBigDecimal().multiply(wd)));
+            updateState(CHANNEL_POWER_RELATIVE,
+                    new DecimalType(millData.get(CHANNEL_POWER_RELATIVE).getAsBigDecimal()));
+            updateState(CHANNEL_ENERGY, new DecimalType(millData.get(CHANNEL_ENERGY).getAsBigDecimal()));
+            updateState(CHANNEL_ENERGY_FC, new DecimalType(millData.get(CHANNEL_ENERGY_FC).getAsBigDecimal()));
+            updateState(CHANNEL_RUNTIME, new DecimalType(millData.get(CHANNEL_RUNTIME).getAsBigDecimal()));
+            updateState(CHANNEL_RUNTIME_PER, new DecimalType(millData.get(CHANNEL_RUNTIME_PER).getAsBigDecimal()));
+            updateState(CHANNEL_LAST_UPDATE, new DateTimeType(millData.get(CHANNEL_LAST_UPDATE).getAsString()));
+
+            if (!getThing().getStatus().equals(ThingStatus.ONLINE)) {
+                updateStatus(ThingStatus.ONLINE);
+            }
+
+        } catch (Exception e) {
+            logger.debug(e.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    private String getMillData() throws IOException {
+
+        String baseURL = "https://zep-api.windcentrale.nl/production/";
+        String urlString = baseURL + millId + "/live?ignoreLoadingBar=true";
+        try {
+            URL url = new URL(urlString);
+            URLConnection connection = url.openConnection();
+            return IOUtils.toString(connection.getInputStream());
+        } catch (MalformedURLException e) {
+            logger.debug("Constructed url '{}' is not valid: {}", urlString, e.getMessage());
+            return null;
```

Comment:
I think that because `null` is returned here `updateData()` will start throwing `NullPointerException`s which are set as status detail for the COMMUNICATION_ERROR?